app/controllers/samples_controller.rb

```diff
@@ -138,14 +139,19 @@ def index_v2
                  current_power.samples
                end
 
-    respond_to do |format|
-      format.json do
-        render json: @samples.offset(offset).limit(limit).as_json(
-          only: [:id, :name, :sample_tissue, :host_genome_id, :project_id, :created_at],
-          methods: []
-        )
-      end
+    @samples = @samples.offset(offset).limit(limit)
+
+    # Efficiently fetch auxiliary information from associated models
+    samples_resp = []
+    @samples.includes(:project, :host_genome, :input_files).each do |s|
+      # TODO: sample_tissue column is deprecated, use Metadatum model instead
+      entry = s.as_json(only: [:id, :name, :sample_tissue, :host_genome_id, :project_id, :created_at])
+      entry["host_genome_name"] = s.host_genome_name # this method uses s.host_genome under the hood
```

Comment:
Since we include/join host_genome above, we should not need to call the `host_genome_name`, right?

Reply:
So the issue is that `host_genome_name` is actually a function, not a column. That function uses `host_genome.name`. So if we don't have `includes(:host_genome)`, the loop will query the host_genomes table in every iteration. If we do `includes(:host_genome)`, then all the records are loaded in one go, so there will be only one query on host_genomes, whose results we access multiple times

Reply:
Yes, i am just saying that since we are doing includes here now we can just call `s.host_genome.name` directly.
Eventually, the `host_genome_name` function can be deleted.

Reply:
That's a good idea